Bump micronaut-data to 3.3.0-RC1

micronaut-build opened the pull request.

micronaut-build committed:
Bump micronaut-data to 3.3.0-RC1

jameskleeh reviewed (changes requested):
@dstepanov This should target 3.4.x yeah? Why was the mysql version erased?

sdelamo committed:
build: set mysql to 8.0.22 same version as micronaut data

Set to https://github.com/micronaut-projects/micronaut-data/blob/master/gradle/libs.versions.toml#L23

sdelamo committed:
build: rever mysql-driver version

dstepanov commented:
No idea why mysql got changed, I’m not sure if we supposed to include this version, before I saw milestones were not present in BOM

sdelamo commented:
This PR should have the base branch in 3.4.x since it is minor version upgrade of data.
